ironfish-rust/src/assets/asset.rs

```diff
@@ -58,23 +54,19 @@ impl Asset {
         let name_bytes = str_to_array(trimmed_name);
         let metadata_bytes = str_to_array(metadata);
 
-        let nonce = 0u8;
-
-        Asset::new_with_raw_data(owner, name_bytes, metadata_bytes, nonce)
+        Asset::new_with_raw_data(owner, name_bytes, metadata_bytes)
     }
 
     fn new_with_raw_data(
         owner: PublicAddress,
         name: [u8; NAME_LENGTH],
         metadata: [u8; METADATA_LENGTH],
-        nonce: u8,
     ) -> Result<Asset, IronfishError> {
-        let capacity = METADATA_LENGTH + NAME_LENGTH + PUBLIC_ADDRESS_SIZE + 1;
+        let capacity = METADATA_LENGTH + NAME_LENGTH + PUBLIC_ADDRESS_SIZE;
```

Comment:
Maybe we can do this to prevent duplicate lengths

```suggestion
        let capacity = ASSET_LENGTH;
```